app/app.py

```diff
@@ -0,0 +1,301 @@
+import os, json, time
+from flask import Flask, request, redirect, render_template, send_from_directory, Response, Markup
+from werkzeug import secure_filename
+import worker 
+import rq
+from rq.registry import StartedJobRegistry
+from db.db import * 
+from os import listdir
+from os.path import isfile, join
+
+
+# Start Flask
+# ----------------------------------------------------------------------------------------------
+app = Flask(__name__, template_folder="templates", static_folder="files", static_url_path="/files")
+app.config['SECRET_KEY'] = 'QFZwHgcXay'
+worker.rq.init_app(app)
+
+# Main Page
+# ----------------------------------------------------------------------------------------------
+@app.route('/', methods=['GET'])
+def index(): 
+    try:
+        default_queue = rq.get_queue()
+        print('default_queue: ', default_queue)
+    except:
+        pass
+
+    return render_template('index.html')
+
+# Ajax for Main Page
+# ----------------------------------------------------------------------------------------------
+@app.route('/last_5_jobs', methods=['GET'])
+def last_5_jobs():
+    table = '''
+        <table class="table table-bordered" style="font-size:12px">
+        <thead>
+            <tr>
+                <th scope="col">pdf_id</th>
+                <th scope="col">filename</th>
+                <th scope="col">status</th>
+                <th scope="col">job_id</th>
+                <th scope="col">received_at</th>
+                <th scope="col">enqueued_at</th>
+                <th scope="col">processed_at</th>
+                <th scope="col">img_count</th>
+            </tr>
+        </thead>
+        <tbody>
+    '''
+    
+    sql = '''
+        select pdf_id
+            , filename
+            , status
+            , job_id
+            , created_at
+            , enqueued_at
+            , processed_at
+            , created_at - processed_at as processing_time
+            ,(select count(1) from imgs where imgs.pdf_id = pdfs.pdf_id) as img_count
+        from pdfs
+        order by created_at desc
+        limit 5
+    '''
+    data = select(sql)
+
+    for pdf_id, filename, status, job_id, created_at, enqueued_at, processed_at, processing_time, img_count in data:
+        table += '''
+            <tr>
+                <th>'''+str(pdf_id)+'''</th>
+                <td>
+                    <a id="pdf_'''+str(pdf_id)+'''" onclick="get_pdf_id_imgs('''+str(pdf_id)+''')" href="" data-bs-toggle="modal" data-bs-target="#modal">
+                    '''+str(filename)+'''
```

Comment:
Ah! I actually had from Werkeug the Secure_Filename when upload but I removed it at a later stage, because it was conflicting and forogt to add it again. This really bugs me. I absolutely agree that this needs to be corrected.

Reply:
If the html would be constructed in a template, then there is a filter for that https://jinja.palletsprojects.com/en/2.11.x/templates/#html-escaping. (Just a note: that function usage is not limited to templates, so it could be used even in here)